### PR TITLE
SfAccordionFeature/sf accordion

### PR DESCRIPTION
**Pull request by michasik**
# Related issue
https://github.com/DivanteLtd/storefront-ui/issues/168

# Scope of work
<code>SfAccordion</code> enables user to toggle visibility of parts of the content.
User has few options how to use the component:
1. Pass <code>items</code> array to populate accordion with predefined styles.
2. Pass multiple <code>SfAccrodionItem</code> into <code>#items</code> slot.
2.1. To use custom styling for header populate <code>#header</code> slot inside <code>SfAccordionItem</code> tag and set <code>showChevron</code> to false in <code>SfAccordion</code> tag.
To use custom styling for opened and closed header states use classes:
        <code>sf-accordion-item__header-slot--close</code>
        <code>sf-accordion-item__header-slot--open</code>
2.2. To use custom styling for content items populate <code>#content</code> slot inside <code>SfAccordionItem</code> tag.

Please have a look on updated stories to see the visual changes and actual usage.

# Screenshots of visual changes
accordion_04

# Checklist

- [x] I followed [composition rules](https://github.com/DivanteLtd/storefront-ui/blob/master/docs/component-rules.md) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/shared/styles/components/SfAccordion.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
+@import '../variables';
+
+$sf-accordion-item__header-padding: 1rem 2rem;
```
Comment (leomp12) [inline packages/shared/styles/components/SfAccordion.scss]:
```suggestion
$accordion-item__header-padding: 1rem 2rem !default;
```
SCSS vars shouldn't have `$sf-` prefix, and you should use `!default` for all component variables

Comment (leomp12) [reply packages/shared/styles/components/SfAccordion.scss]: same for all vars below...

File: packages/vue/src/components/organisms/SfAccordion/SfAccordion.html [hunk 2]
```diff
@@ -1,3 +1,4 @@
-<div>
+<div class="sf-accordion">
+  <!--@slot Default slot for multiple SfAccordionItem elements-->
   <slot></slot>
```
Comment (leomp12) [inline packages/vue/src/components/organisms/SfAccordion/SfAccordion.html]: I think you should have a default markup and not only an empty slot.
My suggestion is receive a array of objects as prop, maybe named `items`, item object may have `label` and `id` properties, then you can render the list of accordion items by default, and emit an event on click passing the id as param.
Wdyt?

Comment (leomp12) [reply packages/vue/src/components/organisms/SfAccordion/SfAccordion.html]: `items` may also have another `items` properties (`items.items`) optionally, then you can handle multi-level

**Comment by filrak** [comment]
@michasik also the gray version shouldn't be a default one (i believe you took inspiration from footer). Could you please add it as css modifier and do background white with previous styling by default?

Backstory: initially this component was used here:
image

File: packages/vue/src/components/organisms/SfAccordion/SfAccordion.js [hunk 3]
```diff
@@ -4,5 +4,66 @@ import SfAccordionItem from "./_internal/SfAccordionItem.vue";
...
+        });
+      } else {
+        const clickedHeader = this.$children.find(child => {
+          return child._uid === slotId;
+        });
+        clickedHeader.isOpen = !clickedHeader.isOpen;
+      }
+    },
+    openFirst() {
+      this.firstOpen ? (this.$children[0].isOpen = true) : "";
```
Comment (dimitor115) [inline packages/vue/src/components/organisms/SfAccordion/SfAccordion.js]: You don't have to use **conditional operator** if you just want to perform some operation, based on some condition. Use `&&` operator instead : `this.firstOpen && (this.$children[0].isOpen = true)`, and you don't have to put this dummy empty string value in negative option.

However for me, still more readable is old fashioned `if`.

File: packages/shared/styles/components/SfAccordion.scss (hunk 1 above)
Comment (michasik) [reply packages/shared/styles/components/SfAccordion.scss]: Thanks, updated :-)

File: packages/vue/src/components/organisms/SfAccordion/SfAccordion.html (hunk 2 above)
Comment (michasik) [reply packages/vue/src/components/organisms/SfAccordion/SfAccordion.html]: Hi @leomp12 I agree with you regarding the first comment, I have implemented it as you described. But I think the idea from the second comment might be a little bit too much for this component. User can always use slots for that imho and implementing that functionality would be an overkill.

File: packages/vue/src/components/organisms/SfAccordion/SfAccordion.js (hunk 3 above)
Comment (michasik) [reply packages/vue/src/components/organisms/SfAccordion/SfAccordion.js]: Hi @dimitor115
In general I agree, but here the easiest way is to just simply:
<code>this.$children[0].isOpen = this.firstOpen</code>. But probably without your comment I would have never refactored this line ;-)

**Comment by michasik** [comment]
@filrak @leomp12 
Hi guys, please have a look and review the changes. Cheers!

File: packages/shared/styles/components/SfAccordion.scss [hunk 4]
```diff
@@ -0,0 +1,67 @@
+@import "../variables";
+
+$accordion-item__header-padding: 0.5rem 0 !default;
+$accordion-header-font-size: 18px;
```
Comment (leomp12) [inline packages/shared/styles/components/SfAccordion.scss]:
```suggestion
$accordion-item__header-font-size: 18px;
```

File: packages/shared/styles/components/SfAccordion.scss [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    &--open {
+      .sf-accordion-item__header-slot--open {
+        display: none;
+      }
+    }
+  }
+  &-item__header {
+    padding: $accordion-item__header-padding;
+    font-size: $accordion-header-font-size;
```
Comment (leomp12) [inline packages/shared/styles/components/SfAccordion.scss]:
```suggestion
    font-size: $accordion-item__header-font-size;
```

File: packages/vue/src/components/organisms/SfAccordion/SfAccordion.js [hunk 6]
```diff
@@ -4,5 +4,64 @@ import SfAccordionItem from "./_internal/SfAccordionItem.vue";
...
+  props: {
+    /**
+     * Array of items to populate the default markup
+     */
+    items: {
+      type: Array,
+      default: () => []
+    },
+    /** Callback function to handle click on accordion item content element */
+    contentClickedCallback: {
```
Comment (leomp12) [inline packages/vue/src/components/organisms/SfAccordion/SfAccordion.js]: imo we don't need this prop, I think is better to just emit a `click` event, wdyt?

File: packages/vue/src/components/organisms/SfAccordion/_internal/SfAccordionItem.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    setActiveElement() {
+      const prevActive = document.getElementsByClassName("sf-accordion-item__content--active")[0];
+      if (prevActive) {
+        prevActive.classList.remove("sf-accordion-item__content--active");
+      }
+      event.target.classList.add("sf-accordion-item__content--active");
+    },
+    onContentItemClick(id) {
+      this.setActiveElement();
+      this.$parent.$emit("contentClicked", id);
```
Comment (leomp12) [inline packages/vue/src/components/organisms/SfAccordion/_internal/SfAccordionItem.js]: why not just `click`?
```suggestion
      this.$parent.$emit("click", id);
```

**Comment by filrak** [comment]
Also resolve the conflicts ;)

**Pull request merged**